### PR TITLE
Add marker API

### PR DESCRIPTION
**Pull request by mravn-google**
* Add marker API
* Simplified example pages
* Consolidated some Java code

### CODE REVIEW

File: packages/google_mobile_maps/android/src/main/java/io/flutter/plugins/googlemobilemaps/GoogleMobileMapsPlugin.java [hunk 1]
```diff
@@ -182,6 +198,10 @@ private static LatLngBounds toLatLngBounds(Object o) {
     return new LatLngBounds(toLatLng(data.get(0)), toLatLng(data.get(1)));
   }
 
+  private static boolean toBoolean(Object o) {
```
Comment (sigurdm) [inline packages/google_mobile_maps/android/src/main/java/io/flutter/plugins/googlemobilemaps/GoogleMobileMapsPlugin.java]: neccesary?

File: packages/google_mobile_maps/android/src/main/java/io/flutter/plugins/googlemobilemaps/GoogleMobileMapsPlugin.java [hunk 2]
```diff
@@ -373,16 +433,57 @@ void hideOverlay() {
...
-    googleMap.addMarker(options);
+  String addMarker(MarkerOptions options) {
+    final Marker marker = googleMap.addMarker(options);
+    markers.put(marker.getId(), marker);
+    return marker.getId();
+  }
+
+  void removeMarker(String markerId) {
+    final Marker marker = marker(markerId);
+    marker.remove();
```
Comment (sigurdm) [inline packages/google_mobile_maps/android/src/main/java/io/flutter/plugins/googlemobilemaps/GoogleMobileMapsPlugin.java]: Do you also want to remove from `markers`?

File: packages/google_mobile_maps/lib/src/bitmap.dart [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  static const BitmapDescriptor defaultMarker =
+      const BitmapDescriptor._(<dynamic>['defaultMarker']);
+
+  static BitmapDescriptor defaultMarkerWithHue(double hue) {
+    assert(0.0 <= hue && hue < 360.0);
+    return new BitmapDescriptor._(<dynamic>['defaultMarker', hue]);
+  }
+
+  static BitmapDescriptor fromAsset(String assetName, [String package]) {
```
Comment (sigurdm) [inline packages/google_mobile_maps/lib/src/bitmap.dart]: Make package a named argument to align with AssetImage.

File: packages/google_mobile_maps/lib/src/bitmap.dart [hunk 4]
```diff
@@ -0,0 +1,48 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of google_mobile_maps;
+
+class BitmapDescriptor {
```
Comment (sigurdm) [inline packages/google_mobile_maps/lib/src/bitmap.dart]: Perhaps worthy of a short dart-doc as the name is very generic.

File: packages/google_mobile_maps/android/src/main/java/io/flutter/plugins/googlemobilemaps/GoogleMobileMapsPlugin.java (hunk 1 above)
Comment (mravn-google) [reply packages/google_mobile_maps/android/src/main/java/io/flutter/plugins/googlemobilemaps/GoogleMobileMapsPlugin.java]: Made use of `toXxx(...)` consistent across all types.

File: packages/google_mobile_maps/android/src/main/java/io/flutter/plugins/googlemobilemaps/GoogleMobileMapsPlugin.java (hunk 2 above)
Comment (mravn-google) [reply packages/google_mobile_maps/android/src/main/java/io/flutter/plugins/googlemobilemaps/GoogleMobileMapsPlugin.java]: Yes, good catch. Done.

File: packages/google_mobile_maps/lib/src/bitmap.dart (hunk 4 above)
Comment (mravn-google) [reply packages/google_mobile_maps/lib/src/bitmap.dart]: Done.

File: packages/google_mobile_maps/lib/src/bitmap.dart (hunk 3 above)
Comment (mravn-google) [reply packages/google_mobile_maps/lib/src/bitmap.dart]: Done.

**Pull request merged**